movai_core_shared/core/message_client.py

```diff
@@ -0,0 +1,85 @@
+"""
+   Copyright (C) Mov.ai  - All Rights Reserved
+   Unauthorized copying of this file, via any medium is strictly prohibited
+   Proprietary and confidential
+
+   Usage:
+        Sends messages to the robot Message-Server.
+
+   Developers:
+   - Ofer Katz (ofer@mov.ai) - 2022
+   - Erez Zomer (erez@mov.ai) - 2022
+"""
+import random
+import time
+
+from movai_core_shared.core.zmq_client import ZMQClient
+from movai_core_shared.exceptions import ArgumentError
+from movai_core_shared.envvars import (
+    MESSAGE_SERVER_LOCAL_ADDR,
+    DEVICE_NAME,
+    FLEET_NAME
+)
+
+
+
+class MessageClient:
+    """
+    This class is the client for message-server.
+    It wraps the data into the message structure and send it to
+    the message-server using ZMQClient.
+    """
+    def __init__(self, msg_type: str) -> None:
+        """
+        constructor - initializes the object.
+
+        Args:
+            msg_type (str): The type of the message, this will affect which
+            handler will be used to handle the request.
+        
+        Raises:
+            TypeError: In case the supplied argument is not in the correct type.
+            ArgumentError: In case the supplied argument is None or an empty string.
+
+        """
+        if not isinstance(msg_type, str):
+            raise TypeError("msg_type argument must be of type string!")
+        if msg_type is None or msg_type == "":
+            raise ArgumentError("msg_type argument must be a non empty string!")
+        self._msg_type = msg_type
+        #TODO: find a solution to import robot id without activating the logger.
+        self._robot_id = str(random.getrandbits(24))
```

Comment:
why we are generating the robot_id?

Reply:
it's not possible to import the robot-id because this class is used inside the logger and the components which retrieve this info require the logger which cause a circular import.
This issue will be fixed on branch 865